marklogic-data-hub/src/main/resources/ml-modules/root/marklogic-data-hub-spark-connector/readLib.sjs

```diff
@@ -170,7 +207,7 @@ function buildSchemaFieldsBasedOnTdeColumns(schemaName, viewName, selectedColumn
     if (selectedColumns == null || selectedColumns.includes(columnName)) {
       schemaFields.push({
         name: columnName,
-        type: column["sys.sys_columns.type"],
+        type: tdeToSparkDataTypeMap.get(column["sys.sys_columns.type"]),
```

Comment:
To make this future-proof (in case TDE introduces a new column type), let's say that if the type value is not in the map, we'll just use string.